_episodes/03-monitor-the-model.md

```diff
@@ -486,79 +518,80 @@ That is a clear signature of overfitting.
 Overfitting is a very common issue and there are many strategies to handle it.
 Most similar to classical machine learning might to **reduce the number of parameters**.
 
-
-We can keep the network architecture unchanged (2 dense layers + a one-node output layer) and only play with the number of nodes per layer.
-
-~~~
-def create_nn(nodes1, nodes2):
-    # Input layer
-    inputs = keras.layers.Input(shape=(X_data.shape[1],), name='input')
-
-    # Dense layers
-    layers_dense = keras.layers.Dense(nodes1, 'relu')(inputs)
-    layers_dense = keras.layers.Dense(nodes2, 'relu')(layers_dense)
-
-    # Output layer
-    outputs = keras.layers.Dense(1)(layers_dense)
-
-    return keras.Model(inputs=inputs, outputs=outputs, name="model_small")
-
-model = create_nn(10, 5)
-~~~
-{:.language-python}
-
-Let's check the created model for good measure:
-
-~~~
-model.summary()
-~~~
-{:.language-python}
-
-~~~
-Model: "model_small"
-_________________________________________________________________
-Layer (type)                 Output Shape              Param #   
-=================================================================
-input (InputLayer)           [(None, 89)]              0         
-_________________________________________________________________
-dense_9 (Dense)              (None, 10)                900       
-_________________________________________________________________
-dense_10 (Dense)             (None, 5)                 55        
-_________________________________________________________________
-dense_11 (Dense)             (None, 1)                 6         
-=================================================================
-Total params: 961
-Trainable params: 961
-Non-trainable params: 0
-
-~~~
-{:.output}
-
-With this change, we have reduced the parameters by 92%. Now compile the model and run the training.
-
-~~~
-model.compile(optimizer='adam',
-              loss='mse',
-              metrics=[keras.metrics.RootMeanSquaredError()])
-history = model.fit(X_train, y_train,
-                    batch_size = 32,
-                    epochs = 200,
-                    validation_data=(X_val, y_val), verbose = 2)
-~~~
-{:.language-python}
-
-In order to compare, we use the same code to check the training performance.
-
-~~~
-history_df = pd.DataFrame.from_dict(history.history)
-sns.lineplot(data=history_df[['root_mean_squared_error', 'val_root_mean_squared_error']])
-plt.xlabel("epochs")
-plt.ylabel("RMSE")
-~~~
-{:.language-python}
-
-![Output of plotting sample](../fig/03_training_history_3_rmse_smaller_model.png)
-
+> ## Try to reduce the degree of overfitting by lowering the number of parameters
```

Comment:
I personally think that the content of this exercise reads more like teaching content. I would object to make it an exercise. Can you comment on why you took that step?

Reply:
You could use a different part of the dataset for the exercise.

Reply:
There is indeed an exercise like that already added in PR #196 . But that's at the very end, as a wrap up exercise, I like that this exercise is about halfway through. My preference is to leave it in as an exercise. I think the exercise can be safely skipped if needed, then there's just the mention of the fact that reducing parameters could work.